portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -97,6 +112,43 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```

Comment:
```suggestion
  @Nullable
  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```
Add @Nullable to make sure callers have null checks.